File: solutions/TypeScript/[1980_Medium] Find Unique Binary String/findDifferentBinaryString.ts
function findDifferentBinaryString(nums: string[]): string {
  nums.sort();

  const n = nums.length;
  let missingNum = 0;

  while (missingNum < n && parseInt(nums[missingNum], 2) === missingNum) {
    missingNum++;
  }

  const missingBinary = missingNum.toString(2);
  const zeroPadLength = n - missingBinary.length;
  return '0'.repeat(zeroPadLength) + missingBinary;
}

File: solutions/Java/[1980_Medim] Find Unique Binary String/findDifferentBinaryString.java
class Solution {
  public String findDifferentBinaryString(String[] nums) {
    Arrays.sort(nums);

    int n = nums.length;
    int missingNum = 0;

    while (missingNum < n && Integer.parseInt(nums[missingNum], 2) == missingNum) {
      missingNum++;
    }

    String missingBinary = Integer.toBinaryString(missingNum);
    StringBuilder builder = new StringBuilder();

    int zeroPadLength = n - missingBinary.length();

    while (zeroPadLength > 0) {
      builder.append('0');
      zeroPadLength--;
    }

    builder.append(missingBinary);

    return builder.toString();
  }
}
